src/parsing.rs

```diff
@@ -110,13 +137,40 @@ fn parse(
       })?;
   let program = post_process(program);
 
+  let (program, top_level_context) = if params.scope_analysis {
+    #[cfg(feature = "transforms")]
+    {
+      use crate::swc::common::Globals;
+      use crate::swc::common::Mark;
+      use crate::swc::common::SyntaxContext;
+      use crate::swc::transforms::resolver::ts_resolver;
+      use crate::swc::visit::FoldWith;
+
+      let globals = Globals::new();
+      crate::swc::common::GLOBALS.set(&globals, || {
+        // This is used to apply proper "syntax context" to all AST elements.
+        let top_level_mark = Mark::fresh(Mark::root());
+        let program = program.fold_with(&mut ts_resolver(top_level_mark));
+        let top_level_context =
+          SyntaxContext::empty().apply_mark(top_level_mark);
+
+        (program, Some(top_level_context))
```

Comment:
It seems like we only need to keep around the `top_level_context` for deno_lint. Turned out to be not too bad at all 🙂